net-p2p/biglybt/biglybt-3.8.0.2.ebuild

```diff
@@ -0,0 +1,128 @@
+#   error: package com.biglybt.ui.swt.plugin.net.buddy does not exist
+# no USE=test due to unsupported testing framework #839681
+JAVA_PKG_IUSE="source"
+MAVEN_PROVIDES="com.${PN}:${PN}-core:${PV} com.${PN}:${PN}-ui:${PV}"
+
+inherit desktop java-pkg-2 java-pkg-simple optfeature xdg
+
+DESCRIPTION="Feature-filled Bittorrent client based on Azureus"
+HOMEPAGE="https://www.biglybt.com"
+SRC_URI="https://github.com/BiglySoftware/BiglyBT/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+S="${WORKDIR}/BiglyBT-${PV}"
+
+LICENSE="GPL-2+"
+SLOT="0"
+KEYWORDS="~amd64 ~arm64 ~ppc64"
+
+SWT_SLOT="4.34"
+CP_DEPEND="
+	dev-java/commons-cli:0
+	dev-java/swt:${SWT_SLOT}[cairo]
+"
+
+# Due to removal of AENameServiceDescriptor it would fail to start under jdk:1.8
+# StartServer ERROR: unable to bind to 127.0.0.1:6880 listening for passed torrent info: \
+# sun.net.spi.nameservice.NameServiceDescriptor: Provider com.biglybt.core.util.spi.AENameServiceDescriptor not found
+#
+# Also it fails with >=jdk-23 so we limit the max version #936549
+# NOTE: BiglyBT works with [headless-awt]
+DEPEND="${CP_DEPEND}
+	|| (
+		virtual/jdk:11
+		virtual/jdk:17
+		virtual/jdk:21
+	)
+"
+RDEPEND="${CP_DEPEND}
+	>=virtual/jre-1.8:*
+"
+
+DOCS=( README.md ChangeLog.txt )
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-3.2.0.0-disable-SWTUpdateChecker.patch
+	"${FILESDIR}"/${PN}-3.2.0.0-disable-shared-plugins.patch
+	"${FILESDIR}"/${PN}-3.5.0.0-Entities.javadoc.patch
+	"${FILESDIR}"/${PN}-3.6.0.0-disable-PluginUpdatePlugin.patch
+	"${FILESDIR}"/${PN}-3.6.0.0-fix-ambiguous.patch
+)
+
+# eclass doc order
+JAVA_GENTOO_CLASSPATH_EXTRA="target/classes"
+JAVA_SRC_DIR="uis/src" # we compile core manually to avoid bloating the jar
+JAVA_RESOURCE_DIRS=( {core,uis}/resources )	# but use resources from both
+JAVA_MAIN_CLASS="com.${PN}.ui.Main"
+JAVA_JAR_FILENAME="BiglyBT.jar"
+
+declare -r ASSET_DIR="${S}"/assets/linux
+src_prepare() {
+	default #780585
+	java-pkg-2_src_prepare
+
+	find -type f -name ".editorconfig" -o -name ".gitignore" -delete \
+		|| die "Cleaning distfile artifacts failed"
+
+	# AENameServiceDescriptor fails to compile with jdk >= 11
+	# "error: package sun.net.spi.nameservice does not exist"
+	# https://github.com/BiglySoftware/BiglyBT/pull/2611
+	rm -r core/src/com/biglybt/core/util/spi || die "deleting spi failed"
+
+	# java-pkg-simple.eclass expects resources in a separate directory
+	# REVIEW: instead of copying all and deleting it would be better to copy selectively
+	for module in core uis; do
+		cp -r ${module}/{src,resources} || die "copying source for resources failed"
+		find ${module}/resources -type f -name "*.java" -delete || die "deleting java files failed"
+	done
+	rm -r core/resources/META-INF || die "deleting copied META-INF failed"
+
+	# patch the desktop file
+	sed -i \
+		-e '/#!/d' \
+		-e 's|${installer:dir.main}/||' \
+		-e 's|.svg||' \
+		"${ASSET_DIR}"/${PN}.desktop || die "patching desktop file failed"
```

Comment:
One more thing, you know `sed` does not `die`?
For this reason Flowdalic asked me to  [introduce eapi9-pipestatus](https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=9f156f818dbe003dcd4989eb7f498895bf580ed1). Not sure if applicable here.